pkg/appmanager/resourceConfig.go

```diff
@@ -2071,25 +2071,57 @@ func (appMgr *Manager) handleRouteTls(
 		rc.MergeRules(appMgr.mergedRulesMap)
 	}
 
-	// Add whitelist conditions to every rule (to prevent access)
+	// Add whitelist condition
 	var whitelistSourceRanges []string
 	if sourceRange, ok := route.ObjectMeta.Annotations[f5VsWhitelistSourceRangeAnnotation]; ok {
 		whitelistSourceRanges = parseWhitelistSourceRangeAnnotations(sourceRange)
 	}
 	if len(whitelistSourceRanges) > 0 {
 		for _, pol := range rc.Policies {
 			if pol.Name == policyName {
-				for _, rl := range pol.Rules {
-					cond := condition{
-						Tcp:     true,
-						Address: true,
-						Matches: true,
-						Name:    "0",
-						Values:  whitelistSourceRanges,
+				for i, rl := range pol.Rules {
+					if rl.FullURI == rule.FullURI && !strings.HasSuffix(rl.Name, "-reset") {
+						origCond := make([]*condition, len(rl.Conditions))
+						copy(origCond, rl.Conditions)
+						cond := condition{
+							Tcp:     true,
+							Address: true,
+							Matches: true,
+							Name:    "0",
+							Values:  whitelistSourceRanges,
+						}
+						if !contains(rl.Conditions, cond) {
+							rl.Conditions = append(rl.Conditions, &cond)
+						}
+
+						// Add reset traffic rule immediately after this rule
+						if (len(pol.Rules) > i+1 && pol.Rules[i+1].Name != rl.Name+"-reset") ||
+							i == len(pol.Rules)-1 {
+							reset := &Rule{
+								Name:    rl.Name + "-reset",
+								FullURI: rl.FullURI,
+								Actions: []*action{{
+									Name:    "0",
+									Forward: true,
```

Comment:
Is forward really required on a reset action?

Reply:
Yep.